include/aspect/material_model/interface.h

```diff
@@ -180,201 +237,216 @@ namespace aspect
 
-      /**
-       * Vector of shared pointers to additional material model input
-       * objects that can be added to MaterialModelInputs. By default,
-       * no inputs are added.
-       */
-      std::vector<std::unique_ptr<AdditionalMaterialInputs<dim> > > additional_inputs;
+        /**
+         * Optional cell object that contains these quadrature
+         * points. This allows for evaluating properties at the cell vertices
+         * and interpolating to the quadrature points, or to query the cell for
+         * material ids, neighbors, or other information that is not available
+         * solely from the locations. Note that not all calling functions will
+         * set this cell iterator. In these cases it will be an invalid iterator
+         * constructed using the default constructor, so make sure that your
+         * material model either fails
+         * with a proper error message, or provides an alternative calculation for
+         * these cases. You can detect this with
+         * @code
+         * if (in.current_cell.state() == IteratorState::valid)
+         * @endcode
+         */
+        typename DoFHandler<dim>::active_cell_iterator current_cell;
 
-      /**
-       * Given an additional material model input class as explicitly specified
-       * template argument, returns a pointer to this additional material model
-       * input object if it is used in the current simulation.
-       * If the output does not exist, a null pointer is returned.
-       */
-      template <class AdditionalInputType>
-      AdditionalInputType *get_additional_input();
+        /**
+         * Vector of shared pointers to additional material model input
+         * objects that can be added to MaterialModelInputs. By default,
+         * no inputs are added.
+         */
+        std::vector<std::unique_ptr<AdditionalMaterialInputs<dim> > > additional_inputs;
 
-      /**
-       * Constant version of get_additional_input() returning a const pointer.
-       */
-      template <class AdditionalInputType>
-      const AdditionalInputType *get_additional_input() const;
+        /**
+         * Given an additional material model input class as explicitly specified
+         * template argument, returns a pointer to this additional material model
+         * input object if it is used in the current simulation.
+         * If the output does not exist, a null pointer is returned.
+         */
+        template <class AdditionalInputType>
+        AdditionalInputType *get_additional_input();
+
+        /**
+         * Constant version of get_additional_input() returning a const pointer.
+         */
+        template <class AdditionalInputType>
+        const AdditionalInputType *get_additional_input() const;
+
+      private:
+        /**
+         * A member variable that stores which properties the material model should compute.
+         */
+        MaterialProperties::Property requested_properties;
```

Comment:
Maybe we should not make this private? All other properties in this class are public (I see it as a struct).

Reply:
I am fine with making it public. I made it private because I thought all use-cases are handled by the request_property function and I find the internal data structure complicated.